builder-golang/action.yaml

```diff
@@ -0,0 +1,66 @@
+name: "SLSA builder for golang"
+description: "Build a binary for a golang project"
+author: "GOSST Team"
+
+# Documentation: https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions.
+
+inputs:
+  go-version:
+    description: "INPUT: Go version to use"
+    required: true
+
+  repo-token:
+    description: "INPUT: GitHub token with read access"
+    required: false
+    default: ${{ github.token }}
+
+#TODO(img)
+branding:
+  icon: "mic"
+  color: "white"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout the repository
+      uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.3.4
+      with:
+        fetch-depth: 0
+        token: ${{ inputs.repo-token }}
+    
+    # Code needed (not tested) if go-vesion is made optional.
+    # See https://stackoverflow.com/questions/67368005/check-whether-environment-variable-is-empty.
+    - name: Set up golang environment
+      uses: actions/setup-go@424fc82d43fa5a37540bae62709ddcc23d9520d4 # v2.1.3
+      with:
+        go-version: ${{ inputs.go-version }}
+    
+    # TODO(hermeticity)
+    # - name: Enable hermeticity
+    # uses: slsa/hermeticity@xxx
+    #   with:
+    #     to-state: enabled
+
+    - name: Download dependencies
+      shell: bash
+      run: |
+        go get -v -d
+
+    # TODO(hermeticity)
+    # - name: Disable hermeticity
+    # uses: slsa/hermeticity@xxx
+    #   with:
+    #     to-state: disabled
+
+    - name: Build project
+      shell: bash
+      run: |
+        go build -o SLSA_BINARY
+
+    #TODO(upload): verify how the action handles malicious names.
+    - name: Upload the artifact
+      uses: actions/upload-artifact@82c141cc518b40d92cc801eee768e7aafc9c2fa2 # v2.3.1
+      with:
+        name: SLSA_BINARY
```

Comment:
should this be indicated as in the `outputs` metadata? Otherwise, there is an implicit interface with whoever is using the action.